src/main/java/frc/robot/subsystems/intake/commands/IntakeCargo.java

```diff
@@ -0,0 +1,51 @@
+package frc.robot.subsystems.intake.commands;
+
+import edu.wpi.first.wpilibj2.command.CommandBase;
+import frc.robot.Constants;
+import frc.robot.subsystems.intake.Intake;
+
+import java.util.function.BooleanSupplier;
+
+/**
+ * Intakes only Cargo colored as the robot's alliance color.
+ */
+public class IntakeCargo extends CommandBase {
+    private final Intake intake;
+    private final BooleanSupplier isAllianceColor;
+    private final double power;
+
+    public IntakeCargo(Intake intake, BooleanSupplier isAllianceColor, double power) {
+        this.intake = intake;
+        this.isAllianceColor = isAllianceColor;
+        this.power = power;
+        addRequirements(intake);
+    }
+
+    @Override
+    public void initialize() {
+        intake.openRetractor();
+    }
+
+    /**
+     * If ball is alliance color, ball will be sucked.
+     */
+    @Override
+    public void execute() {
+        if (isAllianceColor.getAsBoolean())
```

Comment:
Consider adding an `else` clause which will eject unwanted Cargo. This option may be passed as a parameter for this command for extra flexibility.

Reply:
But when this command is operational, the ball is outside of the robot.

Reply:
Could you please rephrase that? I'm not sure I understand what you mean.
Also, something I just realized: if you don't intake Cargo, `isAllianceColor.getAsBoolean() == false` and the intake will never move.

Reply:
I'm saying that when we want to intake cargo, it will be outside of the robot.

Reply:
Also the `isAllianceColor` is a supplier from the camera, not the color sensor.